tests/test_utils.py

```diff
@@ -15,3 +18,31 @@
 )
 def test_standardize_metric_name(framework, logged, standardized):
     assert standardize_metric_name(logged, framework) == standardized
+
+
+class TestConvertDatapointsToListOfDicts:
```

Comment:
Please see if you can use `parametrize` here for these four tests.
Also there will be no need to put them into a class after that, as they will drop to just 2 tests: one for valid outputs and other for invalid outputs.